File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Fixnum
  def divides?(numerator)
    numerator.remainder(self).zero?
  end
end

class Raindrops
  def self.convert(number)
    raindrops = ""

    if 3.divides?(number)
      raindrops += "Pling"
    end

    if 5.divides?(number)
      raindrops += "Plang"
    end

    if 7.divides?(number)
      raindrops += "Plong"
    end

    if raindrops.empty?
      "#{number}"
    else
      raindrops
    end
  end
end

File: r/raindrops/raindrops.R
sounds <- c(Pling = 3, Plang = 5, Plong = 7)

raindrops <- function(number) {
    output <- paste0(names(sounds[number%%sounds == 0]), collapse = "")
    if (nchar(output) != 0) {
        output
    } else {
        as.character(number)
    }
}
